sw/device/tests/power_virus_systemtest.c

```diff
@@ -1150,6 +1154,44 @@ static void max_power_task(void *task_parameters) {
     CHECK_ARRAYS_EQ(received_uart_data, kUartMessage, num_uart_rx_bytes);
   }
 
+  // Check I2C transactions. Only for DVsim
+  if (kDeviceType == kDeviceSimDV) {
+    uint8_t tx_fifo_lvl, rx_fifo_lvl;
+
+    // Make sure all fifo entries have been drained.
+    for (uint32_t ii = 0; ii < ARRAYSIZE(i2c_handles); ++ii) {
+      do {
+        CHECK_DIF_OK(dif_i2c_get_fifo_levels(i2c_handles[ii], &tx_fifo_lvl,
+                                             &rx_fifo_lvl, NULL, NULL));
+      } while (tx_fifo_lvl > 0);
+    };
+
+    // Read data from i2c device.
+    for (uint32_t ii = 0; ii < ARRAYSIZE(i2c_handles); ++ii) {
+      i2c_testutils_rd(i2c_handles[ii], ii + 1, I2C_PARAM_FIFO_DEPTH - 1);
+    };
+
+    // Make sure all data has been read back.
+    for (uint32_t ii = 0; ii < ARRAYSIZE(i2c_handles); ++ii) {
+      do {
+        CHECK_DIF_OK(dif_i2c_get_fifo_levels(i2c_handles[ii], &tx_fifo_lvl,
```

Comment:
same here: `&tx_fifo_level` can be `NULL` instead

Reply:
Done :+1: